scg-system/src/test/java/io/telicent/TestInitialCompaction.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.jena.sparql.core.DatasetGraph;
+import org.apache.jena.tdb1.TDB1Factory;
+import org.apache.jena.tdb1.base.file.Location;
+import org.apache.jena.tdb2.DatabaseMgr;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.stubbing.Answer;
+
+import java.util.Set;
+
+import static io.telicent.TestSmartCacheGraphIntegration.launchServer;
+import static org.apache.jena.graph.Graph.emptyGraph;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.*;
+
+public class TestInitialCompaction {
+    private static final MockedStatic<DatabaseMgr> mockDatabaseMgr  = mockStatic(DatabaseMgr.class, Mockito.CALLS_REAL_METHODS);
+    private static FusekiServer server;
+
+    @BeforeAll
+    static void createAndSetupServerDetails() throws Exception {
+        LibTestsSCG.setupAuthentication();
+        FusekiLogging.setLogging();
+        Attributes.buildStore(emptyGraph);
+    }
+
+    @AfterEach
+    void clearDown() {
+       mockDatabaseMgr.clearInvocations();
+    }
+
+    @AfterAll
+    static void stopServer() throws Exception {
+        if (server != null) {
+            server.stop();
+        }
+        LibTestsSCG.teardownAuthentication();
+        mockDatabaseMgr.close();
+    }
+
+    @Test
+    public void test_emptyGraph() {
+        // given, when
+        server = SmartCacheGraph.construct("--port=0", "--empty").start();
+        // then
+        assertNotNull(server.serverURL());
+        mockDatabaseMgr.verify(() -> DatabaseMgr.compact(any(), anyBoolean()), times(0));
+    }
+
+    @Test
+    public void test_authMemoryDataset() {
+        // given
+        String configFile = "config-no-hierarchies.ttl";
+        // when
+        server = launchServer(configFile);
+        // then
+        assertNotNull(server.serverURL());
+        mockDatabaseMgr.verify(() -> DatabaseMgr.compact(any(), anyBoolean()), times(0));
+    }
+
+    @Test
+    public void test_persistentDataset() {
+        // given
+        mockDatabaseMgr.when(() -> DatabaseMgr.compact(any(),anyBoolean()))
+                .thenAnswer((Answer<Void>) invocation -> null);
+        String configFile = "config-persistent.ttl";
+        // when
+        server = launchServer(configFile);
+        // then
+        assertNotNull(server.serverURL());
+        mockDatabaseMgr.verify(() -> DatabaseMgr.compact(any(), anyBoolean()), times(2));
+    }
+
+    @Test
+    public void test_name() {
+        // given
+        FMod_InitialCompaction fModInitialCompaction = new FMod_InitialCompaction();
+        // when
+        String name = fModInitialCompaction.name();
+        // then
+        assertNotNull(name);
+    }
+
+    @Test
+    public void test_TDB1_DSG() {
```

Comment:
Could maybe do with a few more test cases like this that test that different datasets we commonly expect e.g. `DatasetGraphABAC` are correctly unwrapped for compaction when backed by TDB2?

Reply:
Yeah, just laziness on my part as I could see it was getting called by other tests by the coverage checks. 
I can add them

Reply:
Are there any other common datasets used outside of DatasetGraphABAC?

Reply:
Not for our usage of Jena